src/main/java/org/jabref/gui/util/DefaultTaskExecutor.java

```diff
@@ -19,9 +21,27 @@
 
     private static final Log LOGGER = LogFactory.getLog(DefaultTaskExecutor.class);
 
+    private ExecutorService executor = Executors.newFixedThreadPool(5);
+
+    public static <V> V runInJavaFXThread(Callable<V> callable) {
+        FutureTask<V> task = new FutureTask<>(callable);
+        Platform.runLater(task);
+        try {
+            return task.get();
+        } catch (InterruptedException | ExecutionException e) {
+            LOGGER.error(e);
+            return null;
+        }
+    }
+
     @Override
     public <V> void execute(BackgroundTask<V> task) {
-        new Thread(getJavaFXTask(task)).start();
+        executor.submit(getJavaFXTask(task));
+    }
+
+    @Override
+    public void shutdown() {
+        executor.shutdown();
```

Comment:
Since this is about gui-related task, I assume it would be ok to try to kill them when shutting down instead of waiting until they are finished. Therefore, you should use `shutdownNow()` here, which will interrupt tasks that are executing at the moment of the invocation of the method.